phases/19-capstone-projects/17-personal-ai-tutor/code/main.py

```
@@ -0,0 +1,203 @@
+
+    print("=== 2-week efficacy study (simulated) ===")
+    print(f"curriculum: {len(cmap)} concepts")
+
+    adaptive_gains: list[float] = []
+    baseline_gains: list[float] = []
+    n_learners = 10
+    n_turns = 60
+
+    for i in range(n_learners):
+        ability = rng.gauss(0.3, 0.4)
+        s1 = run_adaptive(f"adapt_{i}", ability, cmap, n_turns, random.Random(100 + i))
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)